src/ApiHandler.ts

```diff
@@ -345,13 +352,101 @@ export default class ApiHandler {
 		}
 	}
 
+	/**
+	 * Fetch and derive generalized staking information at a specific block.
+	 *
+	 * @param hash BlockHash to make call at.
+	 */
+	async fetchStakingInfo(hash: BlockHash): Promise<StakingInfo> {
+		const api = await this.ensureMeta(hash);
+
+		const [
+			validatorCount,
+			forceEra,
+			eraElectionStatus,
+			validators,
+			{ number },
+		] = await Promise.all([
+			await api.query.staking.validatorCount.at(hash),
+			await api.query.staking.forceEra.at(hash),
+			await api.query.staking.eraElectionStatus.at(hash),
+			await api.query.session.validators.at(hash),
+			await api.rpc.chain.getHeader(hash),
+		]);
+
+		const queuedElected = await this.tryFetchQueueElected(api, hash);
+
+		const {
+			eraLength,
+			eraProgress,
+			sessionLength,
+			sessionProgress,
+			activeEra,
+		} = await this.deriveSessionAndEraProgress(api, hash);
+
+		const unappliedSlashesAtActiveEra = activeEra
+			? await api.query.staking.unappliedSlashes.at(hash, activeEra)
+			: null;
+
+		const currentBlockNumber = number.toBn();
+
+		const nextSession = sessionLength
+			.sub(sessionProgress)
+			.add(currentBlockNumber);
+
+		// when ForceNone we are in PoA
+		const nextEra = forceEra.isForceNone
+			? null // If forceEra.isForceNone there is no next era (null)
+			: forceEra.isForceAlways
```

Comment:
coding for a year or so mostly in rust, I've forgotten that this `? :` syntax even exists. I still think if else is more expressive though.

Reply:
(you seem to use it a lot, nevermind then)

Reply:
I prefer it in this case because it allows me to avoid using an undefined `let` over an if/else if/else block.